Documentation/cloud-config.md

```diff
@@ -26,7 +26,7 @@ We've designed our implementation to allow the same cloud-config file to work ac
 
 The cloud-config file uses the [YAML][yaml] file format, which uses whitespace and new-lines to delimit lists, associative arrays, and values.
 
-A cloud-config file must contain `#cloud-config`, followed by an associative array which has zero or more of the following keys:
+A cloud-config file must contain `#cloud-config` header or header starts on `#!`. If cloud-config has `#cloud-config` header, it should followed by an associative array which has zero or more of the following keys:
```

Comment:
This reads a bit confusing to me. What about:

"A cloud-config file must contain a header: either `#cloud-config` for processing as cloud-config (suggested) or `#!` for processing as a shell script (advanced). If cloud-config has `#cloud-config` header, it should followed by an associative array which has zero or more of the following keys:"